src/AuthContextProvider.tsx

```diff
@@ -0,0 +1,124 @@
+import { createUserWithEmailAndPassword, signInWithEmailAndPassword, signOut, GoogleAuthProvider, signInWithPopup } from 'firebase/auth';
+import { auth } from './firebase'; // путь к firebase.ts
+import React from 'react';
+
+interface AuthState {
+  user: string | null;
+  token: string | null;
```

Comment:
Do you use toke anywhere? If not, you don't need to store it.

If the user is not authenticated, you should not have uid in the AuthState. Therefore isAuthenticated prop is redundant - remove.